syntax = "proto3";
import "google/protobuf/empty.proto";

service FileService {
    rpc isConnected(Dummy) returns (google.protobuf.Empty);
    rpc readData(Blockinfo) returns (Data); 
    rpc writeData(Data) returns (google.protobuf.Empty);
    rpc writeDataSpecific(DataSpecific) returns (google.protobuf.Empty);
    rpc deleteFd(File) returns (google.protobuf.Empty);


}
message File {
    int32 fd = 1;
    int64 filesize = 2;
}


message Dummy {
    int32 dummy = 1;
}

message Data {
    bytes data = 1;
    int32 fd = 2;
    int64 blocknum = 3;
}

message DataSpecific {
    bytes data = 1;
    int32 fd = 2;
    int64 blocknum = 3;
    int64 startoffset = 4;
    int64 size = 5;
}

message Blockinfo {
    int32 fd = 1;
    int64 blocknum = 2;
}
